release-23.2: bump version.txt to v23.2.3

celiala opened the pull request:
Release note: None
Epic: None
Release justification: non-production (release infra) change.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/119554)
<!-- Reviewable:end -->

celiala committed:
release-23.2: bump version to v23.2.3

Release note: None
Epic: None
Release justification: non-production (release infra) change.
